content/logs/languages/python.md

```diff
@@ -0,0 +1,140 @@
+---
+title: Python log collection
+kind: documentation
+further_reading:
+- link: "/logs/processing"
+  tag: "Documentation"
+  text: Learn how to process your logs
+- link: "/logs/parsing"
+  tag: "Documentation"
+  text: Learn more about parsing
+- link: "/logs/explore"
+  tag: "Documentation"
+  text: Learn how to explore your logs
+- link: /logs/faq/log-collection-troubleshooting-guide
+  tag: "FAQ"
+  text: Log Collection Troubleshooting Guide
+---
+
+<div class="alert alert-info">
+Datadog's Logs is currently available via public beta. You can apply for inclusion in the beta via <a href="https://www.datadoghq.com/log-management/">this form</a>.
+</div>
+
+## Overview
+
+Use your favorite python logger to log into a file on your host. Then [monitor this file with your Datadog agent](/logs/languages/python/#configure-the-Datadog-agent) to send your logs to Datadog.
+
+## Setup
+Python logs are quite complex to handle, mainly because of tracebacks. They are split into multiple lines which makes them difficult to associate to the original log event.
+To address this use case we strongly recommend to use a JSON formatter when logging in order to:
+
+* Ensure to have a stack_trace properly wrapped into the proper Log
+* Ensure that all the attributes of a log event are properly extracted (severity, logger name, thread name, etc…)
+ 
+Here are setup examples for the two following logging libraries:
+
+- [JSON-log-formatter](https://pypi.python.org/pypi/JSON-log-formatter/0.1.0)
+- [Python-json-logger](https://github.com/madzak/python-json-logger)
+
+### Log into a file
+
+Usage example with [JSON-log-formatter](https://pypi.python.org/pypi/JSON-log-formatter/0.1.0):
+
+```python
+import logging
+
+import json_log_formatter
+
+formatter = json_log_formatter.JSONFormatter()
+
+json_handler = logging.FileHandler(filename='/var/log/my-log.json')
+json_handler.setFormatter(formatter)
+
+logger = logging.getLogger('my_json')
+logger.addHandler(json_handler)
+logger.setLevel(logging.INFO)
+
+logger.info('Sign up', extra={'referral_code': '52d6ce'})
+```
+
+The log file will contain the following log record (inline).
+```json
+{
+    "message": "Sign up",
+    "time": "2015-09-01T06:06:26.524448",
+    "referral_code": "52d6ce"
+}
+```
+
+Usage example with [Python-json-logger](https://github.com/madzak/python-json-logger):
+
+```python
+    import logging
+    from pythonjsonlogger import jsonlogger
+
+    logger = logging.getLogger()
+
+    logHandler = logging.StreamHandler()
+    formatter = jsonlogger.JsonFormatter()
+    logHandler.setFormatter(formatter)
+    logger.addHandler(logHandler)
+```
+ 
+Once the [handler configured](https://github.com/madzak/python-json-logger#customizing-fields), the log file contains the following log record (inline):
```

Comment:
s/handler configured/handler is configured/

Reply:
👍